core/iwasm/common/wasm_memory.c

```diff
@@ -921,7 +921,7 @@ wasm_enlarge_memory_internal(WASMModuleInstance *module, uint32 inc_page_count)
 
 bool
 wasm_runtime_enlarge_memory(WASMModuleInstanceCommon *module_inst,
-                            uint32_t inc_page_count)
+                            uint64 inc_page_count)
 {
```

Comment:
please reject inc_page_count > UINT32_MAX rather than truncating silently.

Reply:
@yamt: I have done this, but it doesn't protect against `cur_page_count + inc_page_count > UINT32_MAX`, so is really not protecting much at all, and will mean that the person who works on true 64 bit support will need to remove this check.

Reply:
> but it doesn't protect against `cur_page_count + inc_page_count > UINT32_MAX`,

really?

i thought this logic was about it. https://github.com/bytecodealliance/wasm-micro-runtime/blob/65bf04e98643f9cbccd6cdc5b8e6d82029c52c38/core/iwasm/common/wasm_memory.c#L789-L792

Reply:
@yamt: oh good, I missed that. So then the overflow isn't an issue, just the need to remove this check in the future.